res/controllers/Stanton-DJC-4-scripts.js

```diff
@@ -0,0 +1,359 @@
+///////////
+// Code. //
+///////////
+
+// ----------   Global variables    ----------
+
+// MIDI Reception commands (from spec)
+djc4.leds = {
+    loopminus: 2,
+    loopplus: 3,
+    loopin: 4,
+    loopout: 5,
+    loopon: 6,
+    loopdel: 7,
+    hotcue1: 8,
+    hotcue2: 9,
+    hotcue3: 10,
+    hotcue4: 11,
+    sample1: 12,
+    sample2: 13,
+    sample3: 14,
+    sample4: 15,
+    keylock: 16,
+    sync: 18,
+    pbendminus: 19,
+    pbendplus: 20,
+    scratch: 21,
+    tap: 22,
+    cue: 23,
+    play: 24,
+    highkill: 25,
+    midkill: 26,
+    lowkill: 27,
+    pfl: 28,
+    fxon: 30,
+    fxexf1: 31,
+    fxexf2: 32,
+    fxexf3: 33,
+    loadac: 34,
+    loadbd: 35,
+    videofx: 36,
+    xflink: 37,
+    keyon: 38,
+    filteron: 39,
+    tx: 46,
+    fx: 47
+};
+
+// ----------   Functions    ----------
+
+// Called when the MIDI device is opened & set up.
+djc4.init = function() {
+    var i;
+
+    // Put all LEDs to default state.
+    djc4.allLed2Default();
+
+    engine.makeConnection("[Channel3]", "track_loaded", djc4.autoShowDecks);
+    engine.makeConnection("[Channel4]", "track_loaded", djc4.autoShowDecks);
+
+    if (engine.getValue("[Master]", "num_samplers") < 8) {
+        engine.setValue("[Master]", "num_samplers", 8);
+    }
+
+    djc4.browseEncoder = new components.Encoder({
+        group: "[Library]",
+        inKey: "Move",
+        input: function(channel, control, value) {
+            if (value === 0x41) {
+                engine.setParameter(this.group, this.inKey + "Down", 1);
+            } else if (value === 0x3F) {
+                engine.setParameter(this.group, this.inKey + "Up", 1);
+            }
+        },
+        unshift: function() {
+            this.inKey = "Move";
+        },
+        shift: function() {
+            this.inKey = "Scroll";
+        },
+    });
+
+    djc4.deck = [];
+    for (i = 0; i < 4; i++) {
+        djc4.deck[i] = new djc4.Deck(i + 1);
+        djc4.deck[i].setCurrentDeck("[Channel" + (i + 1) + "]");
+    }
+
+    djc4.effectUnit = [];
+    for (i = 0; i <= 3; i++) {
+        djc4.effectUnit[i] = new components.EffectUnit([i + 1]);
+        djc4.effectUnit[i].shiftOffset = 0x32;
+        djc4.effectUnit[i].shiftControl = true;
+        djc4.effectUnit[i].enableButtons[1].midi = [0x90 + i, 0x1F];
+        djc4.effectUnit[i].enableButtons[2].midi = [0x90 + i, 0x20];
+        djc4.effectUnit[i].enableButtons[3].midi = [0x90 + i, 0x21];
+        djc4.effectUnit[i].effectFocusButton.midi = [0x90 + i, 0x1D];
+        djc4.effectUnit[i].knobs[1].midi = [0xB0 + i, 0x09];
+        djc4.effectUnit[i].knobs[2].midi = [0xB0 + i, 0x0A];
+        djc4.effectUnit[i].knobs[3].midi = [0xB0 + i, 0x0B];
+        djc4.effectUnit[i].dryWetKnob.midi = [0xB0 + i, 0x08];
+        djc4.effectUnit[i].dryWetKnob.input = function(channel, control, value) {
+            if (value === 0x41) {
+                // 0.05 is an example. Adjust that value to whatever works well for your controller.
```

Comment:
I'd put this into a variable to tweakables so it can be changed more easily globally.